api/environments/models.py

```diff
@@ -358,6 +367,95 @@ def get_feature_state(
             )
         )
 
+    def get_identity_overrides_queryset(self) -> QuerySet[FeatureState]:
+        ids = self._get_active_feature_states_ids(
+            "identity_id",
+            {"identity__isnull": False, "feature_segment__isnull": True},
+        )
+        result: QuerySet[FeatureState] = FeatureState.objects.filter(id__in=ids)
+        return result
+
+    def _get_active_feature_states_ids(
+        self,
+        extra_group_by_fields: Literal["identity_id"] | None = None,
+        filter_kwargs: dict[str, typing.Any] | None = None,
+    ) -> list[int]:
+        base_qs = FeatureState.objects.get_live_feature_states(
+            environment=self,
+            **(filter_kwargs or {}),
+        )
+
+        group_fields = ["feature_id"]
+        if extra_group_by_fields is not None:
+            group_fields.append(extra_group_by_fields)
+
+        return list(
+            base_qs.values(*group_fields)
+            .annotate(id=Max("id"))
+            .values_list("id", flat=True)
+        )
+
+    def get_features_metrics_queryset(self) -> QuerySet[FeatureState]:
+        ids = self._get_active_feature_states_ids(
+            None,
+            {"identity__isnull": True, "feature_segment__isnull": True},
+        )
+        result: QuerySet[FeatureState] = FeatureState.objects.filter(id__in=ids)
+        return result
+
+    def _get_latest_segment_state_ids_subquery(self) -> list[int]:
+        feature_states_qs = FeatureState.objects.get_live_feature_states(
+            environment=self,
+            additional_filters=Q(
+                identity_id__isnull=True,
+                feature_segment_id__isnull=False,
+            ),
+        )
+
+        fs_base_query = (
+            feature_states_qs.values("feature_id", "feature_segment_id")
+            .annotate(latest_id=Max("id"))
```

Comment:
Why are we doing Max?

Reply:
My bad, thanks for catching up, it's a (useless) artifact from before using get_live_feature_states